src/arm64/src/lj_str_hash_arm64.h

```diff
@@ -0,0 +1,271 @@
+/*
+ * This file defines string hash function using CRC32. It takes advantage of
+ * Arm64 hardware support (crc32 instruction) to speedup the CRC32
+ * computation. The hash functions try to compute CRC32 of length and up
+ * to 128 bytes of given string.
+ */
+
+#ifndef _LJ_STR_HASH_ARM64_H_
+#define _LJ_STR_HASH_ARM64_H_
+
+#if defined(__aarch64__) && defined(__GNUC__)
+
+#include <stdint.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <time.h>
+#include <sys/auxv.h>
+#include <stdio.h>
+#include <arm_acle.h>
+
+#include "../../lj_def.h"
+
+#ifndef HWCAP_CRC32
+#define HWCAP_CRC32             (1 << 7)
+#endif /* HWCAP for crc32 */
+
+#ifndef LJ_AINLINE
+#define LJ_AINLINE   inline __attribute__((always_inline))
+#endif
+
+#ifdef __MINGW32__
+#define random()  ((long) rand())
+#define srandom(seed)  srand(seed)
+#endif
+
+extern uint32_t lj_str_original_hash(const char *str, size_t lenx);
+static LJ_AINLINE uint32_t lj_str_hash(const char* str, size_t len);
+/* lj_str hash function determined at runtime */
+typedef uint32_t (*lj_str_hash_func)(const char *str, size_t lenx);
+lj_str_hash_func LJ_STR_HASH;
+
+static const uint64_t* cast_uint64p(const char* str)
+{
+  return (const uint64_t*)(void*)str;
+}
+
+static const uint32_t* cast_uint32p(const char* str)
+{
+  return (const uint32_t*)(void*)str;
+}
+
+static LJ_AINLINE uint32_t lj_str_hash_1_4(const char* str, uint32_t len)
+{
+  uint32_t v = str[0], h = 0;
+  v = (v << 8) | str[len >> 1];
+  v = (v << 8) | str[len - 1];
+  v = (v << 8) | len;
+  return __crc32cw(h, v);
+}
+
```

Comment:
IIRC, on X86-64, if the string being hashed having length under 4, original hash function seem to be working better.

Other than that, LGTM.

Reply:
As per my tests on ARM64 platform the micro benchmark did not show any noticeable differences for below 4 bytes.